### PR TITLE
feat(TUI-199): add rowRenderers to the VirtualizedList

### PR DESCRIPTION
**Pull request by jmfrancois**
**What is the problem this PR is trying to solve?**

Currently you can t add custom renderer to the VirtualizedList
You can t override the Large display mode in the List component.

**What is the chosen solution to this problem?**

* support `rowRenderers` props on VirtualizedList
* pass `rowRenderers` from List to VirtualizedList

add example in both storybook
add tests in VirtualizedList

**Please check if the PR fulfills these requirements**

* [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
* [x] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
* [x] Docs have been added / updated (for bug fixes / features)
* [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/components/src/List/List.component.js [hunk 1]
```diff
@@ -58,7 +58,6 @@ ListToolbar.propTypes = {
 		}),
 	}),
 	toolbar: PropTypes.shape(omit(Toolbar.propTypes, 't')),
-	renderers: PropTypes.object,
```
Comment (jmfrancois) [inline packages/components/src/List/List.component.js]: previously used but not anymore used anywhere ...

File: packages/components/src/VirtualizedList/RendererSelector.component.js [hunk 2]
```diff
@@ -15,10 +15,12 @@ const { TABLE } = listTypes;
  * Select the ListGrid row renderer to use
  * @param type The row renderer type
  */
-function getRowRenderer(type) {
-	const rowRenderer = rowDictionary[type];
+function getRowRenderer(type, renderers = {}) {
+	debugger;
```
Comment (nicomaligne) [inline packages/components/src/VirtualizedList/RendererSelector.component.js]: Cough cough :)

File: output/components.eslint.txt [hunk 3]
```diff
@@ -8,8 +8,15 @@ The react/require-extension rule is deprecated. Please use the import/extensions
...
+  4:16  error  'LARGE' is assigned a value but never used  no-unused-vars
+
+/home/travis/build/Talend/ui/packages/components/src/VirtualizedList/RendererSelector.component.js
+  19:2  error  Unexpected 'debugger' statement  no-debugger
+
 /home/travis/build/Talend/ui/packages/components/src/VirtualizedList/utils/tablerow.js
   33:3  warning  Unexpected console statement  no-console
 
-✖ 3 problems (2 errors, 1 warning)
+✖ 6 problems (5 errors, 1 warning)
```
Comment (nicomaligne) [inline output/components.eslint.txt]: Some fix here

File: packages/components/src/VirtualizedList/RendererSelector.component.js [hunk 4]
```diff
@@ -92,17 +93,19 @@ class RendererSelector extends React.Component {
 			};
 		} else {
 			ListRenderer = ListGrid;
-			customProps = { rowRenderer: getRowRenderer(type) };
+			customProps = { rowRenderer: getRowRenderer(type, this.props.rowRenderers) };
```
Comment (nicomaligne) [inline packages/components/src/VirtualizedList/RendererSelector.component.js]: You are only allowing custom stuff when not in table mode. Later on we'll be nice to let custom the table mode nop ?

File: packages/components/src/VirtualizedList/PropTypes.js [hunk 5]
```diff
@@ -40,5 +38,6 @@ export default {
 	// Current sort direction ('ASC' | 'DESC')
 	sortDirection: PropTypes.oneOf([SORT_BY.ASC, SORT_BY.DESC]),
 	// List type ('TABLE' | 'LARGE' | ...)
-	type: PropTypes.oneOf([TABLE, LARGE]),
+	type: PropTypes.string,
```
Comment (jmfrancois) [inline packages/components/src/VirtualizedList/PropTypes.js]: allow other type to be used (add custom renderer)

File: packages/components/src/VirtualizedList/RendererSelector.component.js (hunk 4 above)
Comment (jmfrancois) [reply packages/components/src/VirtualizedList/RendererSelector.component.js]: The table can already be customized at the cell level AFAIK

**Pull request merged**